src/euphorie/client/browser/training.py

```diff
@@ -285,8 +302,10 @@ def question_intro_url(self):
     @property
     @view_memoize
     def questions(self):
+        training = self.get_or_create_training()
+        question_ids = loads(training.answers).keys()
         survey = self.webhelpers._survey
-        return survey.listFolderContents({"portal_type": "euphorie.training_question"})
+        return [survey.get(q_id) for q_id in question_ids]
```

Comment:
It seems to me you are getting the objects then you are using their id, dump them in to a json and then load the json to get the id and fetch the objects once again.
It seems that this story can be simplified a little.

Reply:
I'm not sure I understand you correctly. I do get the objects with `survey.get(q_id)`, but they are not dumped to json. `questions()` returns a list of objects.

Or do you mean that `get_or_create_training()` fetches the objects? It only does so once when the training is initialized, and I'm changing that to use the catalog, see your other comment.

I could go through all the uses of `self.questions` and check whether they really need the objects, but I wonder whether it's worth the effort.

Reply:
I did straighten it out a little. See afb0b43